### DISCUSSION TITLE
ColonisationStats function

### DISCUSSION THREAD

**Discussion by ellalewi547** [RangeShifter/RangeshiftR-tutorials#51] [Q&A]
Hello! I wondered if you could help me understand how to run the ColonisationStats in cases where I want to reimport the data from my files rather than in my current workspace as the master_parameter object. 

From the details of the function I have attempted the following code, which gives me an error message that my habitat patch file is not a Raster Layer or Raster Stack, however it is shown in my workspace environment as being a Raster Layer: 

`patches1 <- raster(paste0(dirpath,'Inputs/Final habitat patch V7.asc'))
colonisation <- read.table(paste0(dirpath,"Outputs/Batch1_Sim80_Land1_Pop.txt"), header = T)
col <- ColonisationStats(colonisation, patches1, years = numeric(100))`

Would you be able to guide me to where I am going wrong?

Apologies for my incompetence - I'm not the best with coding.

Thank you for your help, 

Ella

**Comment by JetteReeg** [comment]
Hi Ella,
thank you for pointing me to this issue. The error message is indeed a bit misleading. As we shifted from raster to terra within the RangeShiftR package, the function ColonisationStats() expected a SpatRaster object as input for the patch files assuming that users would use the terra package for reading in raster files.
I have adapted the checks now and the newest version now accepts besides the SpatRaster objects from the terra package also  Raster Layer/Raster Stack objects from the raster package.
You need to reinstall RangeShiftR to apply the bugfix:

```
library(Rcpp)
library(devtools)
library(Rdpack)
devtools::install_github("https://github.com/RangeShifter/RangeShiftR-pkg", ref = "main", subdir="RangeShiftR")
```
